File: src/backend/MyAIRunningMate/MyAIRunningMate.Contracts/Aggregates/Responses/ActivityDetailsResponse.cs
namespace MyAIRunningMate.Contracts.Aggregates.Responses;

public record ActivityDetailsResponse(
    Guid ActivityId,
    string GarminActivityId,
    DateTime StartTime,
    double TotalTime,
    double MovingTime,
    double DistanceMetres,
    int BeginningBodyBattery,
    int BeginningBodyPotential,
    int EndingBodyBattery,
    int EndingPotential,
    int? TotalAscent,
    int? TotalDescent,
    int RecoveryTime,
    string ExerciseType,
    string ExerciseSubType,
    string ExerciseName,
    double UserVolumetricOxygenMax,
    int UserMaxHeartRate,
    int UserLactateThresholdHeartRate,
    int UserLactateThresholdPower,
    double UserLactateThresholdSpeed,
    int NumberOfLaps,
    string? Location,
    string? MapPolyline
    );

File: src/frontend/MyAIRunningMate-UI/src/types/aggregates/activityDetailsResponse.ts
export interface ActivityDetailsResponse {
  activity_id: number;
  garmin_activity_id: string;
  start_time: string;
  total_time: number;
  moving_time: number;
  distance_metres: number;
  beginning_body_battery: number;
  beginning_body_potential: number;
  ending_body_battery: number;
  ending_potential: number;
  total_ascent: number | null;
  total_descent: number | null;
  recovery_time: number;
  exercise_type: string;
  exercise_subtype: string;
  exercise_name: string;
  user_volumetric_oxygen_max: number;
  user_max_heart_rate: number;
  user_lactate_threshold_heart_rate: number;
  user_lactate_threshold_power: number;
  user_lactate_threshold_speed: number;
  number_of_laps: number;
  location: string | null;
  map_polyline: string | null;
}